meilisearch/tests/index/test_index.py

```diff
@@ -73,60 +93,71 @@ def test_get_or_create_index(self):
         assert len(documents) == 1
         index_2.delete()
         with pytest.raises(Exception):
-            self.client.get_index(index_3).info()
+            self.client.get_index(index_3)
 
     def test_get_or_create_index_with_primary_key(self):
         """Test get_or_create_index method with primary key"""
         index_1 = self.client.get_or_create_index('books', {'primaryKey': self.index_uid4})
         index_2 = self.client.get_or_create_index('books', {'primaryKey': 'some_wrong_key'})
+        assert index_1.primary_key == self.index_uid4
         assert index_1.get_primary_key() == self.index_uid4
+        assert index_2.primary_key is None
         assert index_2.get_primary_key() == self.index_uid4
+        assert index_2.primary_key == self.index_uid4
         index_1.delete()
 
-    def test_index_info(self):
-        """Tests getting an index's info"""
-        index = self.client.get_index(uid=self.index_uid)
-        response = index.info()
+    def test_index_fetch_info(self):
+        """Tests getting the index info"""
+        index = self.client.index(uid=self.index_uid)
+        response = index.fetch_info()
         assert isinstance(response, object)
-        assert response['uid'] == self.index_uid
-        assert response['primaryKey'] is None
-
-    def test_index_info_with_wrong_uid(self):
-        """Tests getting an index's info in MeiliSearch with a wrong UID"""
-        with pytest.raises(Exception):
-            self.client.get_index(uid='wrongUID').info()
+        assert response.uid == self.index_uid
+        assert response.primary_key is None
+        assert response.primary_key == index.primary_key
+        assert response.primary_key == index.get_primary_key()
+
+    def test_index_fetch_info_containing_primary_key(self):
+        """Tests getting the index info"""
+        index = self.client.index(uid=self.index_uid3)
+        response = index.fetch_info()
+        assert isinstance(response, object)
+        assert response.uid == self.index_uid3
+        assert response.primary_key == 'book_id'
```

Comment:
There is something I'm missing or something confusing here. Shouldn't this `primary_key` be `None`? You didn't specify any primary key (L121), and it was never updated afterward. So either this assert works because the primary key is coming from a different test (which shouldn't be the case, I think the test should behave the same way in an independent call or in a set of tests) or this assert should fail.

Reply:
https://github.com/meilisearch/meilisearch-python/pull/175#discussion_r525020400